internal/mode/static/nginx/config/maps.go

```diff
@@ -21,7 +22,67 @@ func executeMaps(conf dataplane.Configuration) []executeResult {
 	return []executeResult{result}
 }
 
-func buildAddHeaderMaps(servers []dataplane.VirtualServer) []http.Map {
+func executeStreamMaps(conf dataplane.Configuration) []executeResult {
+	maps := createStreamMaps(conf)
+
+	result := executeResult{
+		dest: streamConfigFile,
+		data: helpers.MustExecuteTemplate(mapsTemplate, maps),
+	}
+
+	return []executeResult{result}
+}
+
+func createStreamMaps(conf dataplane.Configuration) []shared.Map {
+	maps := make([]shared.Map, 0)
+	portsToMap := make(map[int32]int)
+
```

Comment:
Let's short circuit here and if tls passthrough servers has length is zero and if it is bail out of the function.